### PR TITLE
Refactor export structure

### LINKED ISSUE #979: MPS files are overwritten when optimization-range = day
There should be 7*2 = 14 MPS files per week. There are only two.

Example : study 021 from examples.

### PR DESCRIPTION
**Pull request by guilpier-code**
This work is in the scope of issue #979 and is a preliminary refactoring work. 

The names of export files **interconnexion.txt**, **Area.txt** and **Variables.txt** don't need to wear a simulation period (which simulation week, year or even day). Thus, these names does not need to be built with the function **getFilenameWithExtension** that automatically add a simulation period hint in the name of an export file.
These files names composition are now very simple.

### CODE REVIEW

File: src/solver/optimisation/opt_construction_matrice_des_contraintes_cas_lineaire.cpp [hunk 1]
```diff
@@ -1238,13 +1234,11 @@ void OPT_ConstruireLaMatriceDesContraintesDuProblemeLineaire(PROBLEME_HEBDO* Pro
     }
 
     // Export structure
-    if (exportStructure)
+    if (doWeExport)
     {
-        OPT_ExportVariables(*study, varname, "variables", "txt", numSpace);
-
-        // TODO : for now empty constraints.txt file needed
```
Comment (guilpier-code) [inline src/solver/optimisation/opt_construction_matrice_des_contraintes_cas_lineaire.cpp:1140]: Export file **constraints.txt** is empty and useless, so I removed it.
If it turns out to be useful at some point, we will add it easily.

File: src/solver/optimisation/opt_export_structure.cpp [hunk 2]
```diff
@@ -65,43 +62,31 @@ const std::initializer_list<std::string>& getNames<ExportStructDict>()
 } // namespace Data
 } // namespace Antares
 
-void OPT_ExportInterco(const Antares::Data::Study& study,
```
Comment (guilpier-code) [inline src/solver/optimisation/opt_export_structure.cpp:68]: Here we used to pass the whole study. This is unclear as we only need its writer.

File: src/solver/optimisation/opt_export_structure.cpp [hunk 3]
```diff
@@ -140,18 +125,14 @@ void OPT_Export_add_variable(std::vector<std::string>& varname,
...
-                         const std::vector<std::string>& varname,
-                         const std::string& fileName,
-                         const std::string& fileExtension,
-                         uint numSpace)
+void OPT_ExportVariables(const Antares::Solver::IResultWriter::Ptr writer,
+                         const std::vector<std::string>& varname)
 {
     Yuni::Clob Flot;
-    auto filename = getFilenameWithExtension(fileName, fileExtension, numSpace);
+    std::string filename = "Variables.txt";
```
Comment (flomnes) [inline src/solver/optimisation/opt_export_structure.cpp]:
```suggestion
    std::string filename = "variables.txt";
```

File: src/solver/optimisation/opt_construction_matrice_des_contraintes_cas_lineaire.cpp [hunk 4]
```diff
@@ -1238,13 +1234,11 @@ void OPT_ConstruireLaMatriceDesContraintesDuProblemeLineaire(PROBLEME_HEBDO* Pro
...
-        std::vector<std::string> conname;
-        OPT_ExportVariables(*study, conname, "constraints", "txt", numSpace);
+        OPT_ExportInterco(study->resultWriter, ProblemeHebdo);
+        OPT_ExportAreaName(study->areas, study->resultWriter);
+        OPT_ExportVariables(study->resultWriter, varname);
```
Comment (flomnes) [inline src/solver/optimisation/opt_construction_matrice_des_contraintes_cas_lineaire.cpp]: [Style] 
Writer = 1st argument, content = 2nd argument

File: src/solver/optimisation/opt_construction_matrice_des_contraintes_cas_lineaire.cpp [hunk 5]
```diff
@@ -92,12 +92,8 @@ void OPT_ConstruireLaMatriceDesContraintesDuProblemeLineaire(PROBLEME_HEBDO* Pro
...
     Study::Ptr study = Study::Current::Get();
     bool exportStructure = ProblemeHebdo->ExportStructure;
-
-    if (exportStructure)
-    {
-        OPT_ExportInterco(*study, ProblemeHebdo, numSpace);
-        OPT_ExportAreaName(*study, ProblemeHebdo, numSpace);
-    }
+    bool firstWeekOfSimulation = ProblemeHebdo->firstWeekOfSimulation;
+    bool doWeExport = exportStructure && firstWeekOfSimulation;
```
Comment (flomnes) [inline src/solver/optimisation/opt_construction_matrice_des_contraintes_cas_lineaire.cpp]: All these `bool` are `const`

File: src/libs/antares/study/fwd.cpp [hunk 6]
```diff
@@ -347,9 +347,9 @@ std::string mpsExportStatusToString(const mpsExportStatus& mps_export_status)
     {
     case mpsExportStatus::NO_EXPORT:
         return "none";
-    case mpsExportStatus::EXPORT_FIRST_OPIM:
+    case mpsExportStatus::EXPORT_FIRST_OPTIM:
         return "optim-1";
-    case mpsExportStatus::EXPORT_SECOND_OPIM:
+    case mpsExportStatus::EXPORT_SECOND_OPTIM:
```
Comment (flomnes) [inline src/libs/antares/study/fwd.cpp:352]: Nobody ever noticed that typo, good job !

**Review by flomnes — CHANGES_REQUESTED** [review]
LGTM

- Minor changes
- Files variable-x-y.txt are not identical

```
omnesflo@gm0winl370:~/short/029 Pumped storage plant - 2/output/20221220-1610eco$ md5sum variables-* | sort
3477bd34fe695d9e80bd31fa5b8ea266  variables-1-7.txt
3477bd34fe695d9e80bd31fa5b8ea266  variables-2-7.txt
3477bd34fe695d9e80bd31fa5b8ea266  variables-3-7.txt
3477bd34fe695d9e80bd31fa5b8ea266  variables-4-7.txt
3477bd34fe695d9e80bd31fa5b8ea266  variables-5-7.txt
68668da7fbb602eba00272eb60ed1c9b  variables-1-9.txt
68668da7fbb602eba00272eb60ed1c9b  variables-2-9.txt
68668da7fbb602eba00272eb60ed1c9b  variables-3-9.txt
68668da7fbb602eba00272eb60ed1c9b  variables-4-9.txt
68668da7fbb602eba00272eb60ed1c9b  variables-5-9.txt
6d91d159878c892dfd61782afd2879e5  variables-1-2.txt
6d91d159878c892dfd61782afd2879e5  variables-2-2.txt
6d91d159878c892dfd61782afd2879e5  variables-3-2.txt
6d91d159878c892dfd61782afd2879e5  variables-4-2.txt
6d91d159878c892dfd61782afd2879e5  variables-5-2.txt
8a87ecc7eabd35fd8ab30636ba853c6f  variables-1-8.txt
8a87ecc7eabd35fd8ab30636ba853c6f  variables-2-8.txt
8a87ecc7eabd35fd8ab30636ba853c6f  variables-3-8.txt
8a87ecc7eabd35fd8ab30636ba853c6f  variables-4-8.txt
8a87ecc7eabd35fd8ab30636ba853c6f  variables-5-8.txt
92409d8058efa6d42f734d758f2149e8  variables-1-6.txt
92409d8058efa6d42f734d758f2149e8  variables-2-6.txt
92409d8058efa6d42f734d758f2149e8  variables-3-6.txt
92409d8058efa6d42f734d758f2149e8  variables-4-6.txt
92409d8058efa6d42f734d758f2149e8  variables-5-6.txt
9afda112d33708ba0fe68365d59eaf6c  variables-1-10.txt
9afda112d33708ba0fe68365d59eaf6c  variables-2-10.txt
9afda112d33708ba0fe68365d59eaf6c  variables-3-10.txt
9afda112d33708ba0fe68365d59eaf6c  variables-4-10.txt
9afda112d33708ba0fe68365d59eaf6c  variables-5-10.txt
a3e7e5e5d7dcf3be5efad3a27e7caf5c  variables-1-4.txt
a3e7e5e5d7dcf3be5efad3a27e7caf5c  variables-2-4.txt
a3e7e5e5d7dcf3be5efad3a27e7caf5c  variables-3-4.txt
a3e7e5e5d7dcf3be5efad3a27e7caf5c  variables-4-4.txt
a3e7e5e5d7dcf3be5efad3a27e7caf5c  variables-5-4.txt
a4679b58d5d7899ccc48f1351ba6d48c  variables-1-11.txt
a4679b58d5d7899ccc48f1351ba6d48c  variables-2-11.txt
a4679b58d5d7899ccc48f1351ba6d48c  variables-3-11.txt
a4679b58d5d7899ccc48f1351ba6d48c  variables-4-11.txt
a4679b58d5d7899ccc48f1351ba6d48c  variables-5-11.txt
a47fd14f2c4a82c37c06774821cfaaab  variables-1-3.txt
a47fd14f2c4a82c37c06774821cfaaab  variables-2-3.txt
a47fd14f2c4a82c37c06774821cfaaab  variables-3-3.txt
a47fd14f2c4a82c37c06774821cfaaab  variables-4-3.txt
a47fd14f2c4a82c37c06774821cfaaab  variables-5-3.txt
bf3f9bd845f77d36f73a01d70ba1f2e8  variables-1-12.txt
bf3f9bd845f77d36f73a01d70ba1f2e8  variables-2-12.txt
bf3f9bd845f77d36f73a01d70ba1f2e8  variables-3-12.txt
bf3f9bd845f77d36f73a01d70ba1f2e8  variables-4-12.txt
bf3f9bd845f77d36f73a01d70ba1f2e8  variables-5-12.txt
d56b5b993a8b2af24d9e12cccb26582c  variables-1-5.txt
d56b5b993a8b2af24d9e12cccb26582c  variables-2-5.txt
d56b5b993a8b2af24d9e12cccb26582c  variables-3-5.txt
d56b5b993a8b2af24d9e12cccb26582c  variables-4-5.txt
d56b5b993a8b2af24d9e12cccb26582c  variables-5-5.txt
e376ce0ed270136fcd035d30a6a87340  variables-1-13.txt
e376ce0ed270136fcd035d30a6a87340  variables-2-13.txt
e376ce0ed270136fcd035d30a6a87340  variables-3-13.txt
e376ce0ed270136fcd035d30a6a87340  variables-4-13.txt
e376ce0ed270136fcd035d30a6a87340  variables-5-13.txt
fc3dcb76140f73a914f43f59c2eda27e  variables-1-1.txt
fc3dcb76140f73a914f43f59c2eda27e  variables-2-1.txt
fc3dcb76140f73a914f43f59c2eda27e  variables-3-1.txt
fc3dcb76140f73a914f43f59c2eda27e  variables-4-1.txt
fc3dcb76140f73a914f43f59c2eda27e  variables-5-1.txt
```

**Comment by guilpier-code** [comment]
> 
blabla

**Comment by guilpier-code** [comment]
About the **variables-\<year\>-\<week\>.txt** : 
these files only differ because of the last piece of information on each line.
This last piece is the hour in the year (1 to 3760).
So **variables-1-1.txt** and **variables-1-2.txt** are different.
But **variables-1-1.txt** and **variables-2-1.txt** are not.

So I recommend that we remove this last piece of information from each line of **variables-\<year\>-\<week\>.txt**, rename it **variables.txt**, and print this file only at the first week of the simulation, just like **interconnexion.txt** and **area.txt**

**Review by flomnes — APPROVED** [review]

**Pull request merged**